main/src/main/java/org/qbicc/main/Main.java

```diff
@@ -442,7 +443,7 @@ void call0(BaseDiagnosticContext initialContext) {
                                 if (llvm) {
                                     builder.addPreHook(Phase.ADD, LLVMIntrinsics::register);
                                 }
-                                builder.addPreHook(Phase.ADD, CoreIntrinsics::register);
+                                builder.addPreHook(Phase.ADD, ctxt -> CoreIntrinsics.register(ctxt, isWasm));
```

Comment:
Changing this can be avoided by instead using `ctxt.getPlatform().getCpu()` where the intrinsic is generated.